### DISCUSSION TITLE
Check URL load window staus and handler timeout or failed request

### DISCUSSION THREAD

**Discussion by Facundo15** [tauri-apps/tauri#7767] [Q&A]
I'm migrating an electron app to Tauri because of the final size it offers.

I have everything built on a nodejs backend and everything works quite well, I just want to be able to control from a splashscreen and know if it could load or not to request to wait a few minutes and wait to load again.

I could easily use the webView message, but I would also like to control from Rust the window when there is a connection loss in the application

The equivalent to electron of what I am looking for would be this:

```
splashscreen.loadFile('./src/splashscreen.html');
  mainWindow.loadURL('http://localhost:3000').then(() => {
      const currentURL = win.webContents.getURL()
      console.log(currentURL)
  }).catch(err => {
      console.log(err)
      // show the error window
  });
```

**Comment by FabianLars** [comment]
There's no such api yet but feel free to create a feature request (as a github issue) for it. Just a warning tho, we do have way less control over the web engine compared to electron so it _may_ not be possible on all platforms 🤔
